### PR TITLE
OpenMPTarget: Update to LLVM/13 .

### PR DESCRIPTION
**Pull request by rgayatri23**
The PR does the following 
1. Updates the CI testing for OpenMPTarget backend to use LLVM/13 .
2. Uses a single pass LLVM installation rather than a 2 pass installation.
3. It disables one unit test which fails correctness with LLVM/13 . 

The PR is dependent on #4456 to avoid a compile time error.

### CODE REVIEW

File: scripts/docker/Dockerfile.openmptarget [hunk 1]
```diff
@@ -52,22 +53,13 @@ RUN LLVM_VERSION=887c7660bdf3f300bd1997dcfd7ace91787c0584 && \
...
-      -DCMAKE_CXX_COMPILER=$LLVM_DIR/bin/clang++ \
-      -DLIBOMPTARGET_NVPTX_COMPUTE_CAPABILITIES=70 \
-    ../openmp && \
-    make -j${NPROC} && \
-    make install && \
-    echo "${LLVM_DIR}/lib" > /etc/ld.so.conf.d/llvm.conf && ldconfig && \
-    rm -rf ${SCRATCH_DIR}
-ENV PATH=${LLVM_DIR}/bin:$PATH
+    rm -rf ${SCRATCH_DIR} \
+ENV PATH=${LLVM_DIR}/bin:$PATH \
```
Comment (dalg24) [inline scripts/docker/Dockerfile.openmptarget:64]:
```suggestion
ENV PATH=${LLVM_DIR}/bin:$PATH
```

File: .jenkins [hunk 2]
```diff
@@ -191,7 +191,7 @@ pipeline {
                         sh '''rm -rf build && mkdir -p build && cd build && \
                               cmake \
                                 -DCMAKE_BUILD_TYPE=RelWithDebInfo \
-                                -DCMAKE_CXX_COMPILER=clang++ \
+                                -DCMAKE_CXX_COMPILER=/opt/llvm/bin/clang++ \
```
Comment (dalg24) [inline .jenkins:194]: Was this change actually necessary?

Comment (rgayatri23) [reply .jenkins]: Not anymore. Resolved the `PATH` issue in the Dockerfile which fixes this issue.

File: algorithms/unit_tests/TestStdAlgorithmsMinMaxElementOps.cpp [hunk 3]
```diff
@@ -202,7 +202,11 @@ void std_algo_min_max_test_verify(Kokkos::pair<IndexType, ValueType> goldPair,
                                   const ItType result,
                                   TestedViewType testedView) {
   // check that iterator is pointing to right element
+  // FIXME_OPENMPTARGET - With LLVM/13 we need PR 4456 to be merged. Else the
+  // following line fails duirng compile time.
+#ifndef KOKKOS_ENABLE_OPENMPTARGET
   EXPECT_EQ(result - KE::begin(testedView), goldPair.first);
+#endif
```
Comment (dalg24) [inline algorithms/unit_tests/TestStdAlgorithmsMinMaxElementOps.cpp:209]:
```
/var/jenkins/workspace/Kokkos/algorithms/unit_tests/TestStdAlgorithmsMinMaxElementOps.cpp:203:50: error: unused parameter 'testedView' [-Werror,-Wunused-parameter]
                                  TestedViewType testedView) {

File: algorithms/unit_tests/TestStdAlgorithmsMinMaxElementOps.cpp [hunk 4]
```diff
@@ -202,7 +202,11 @@ void std_algo_min_max_test_verify(Kokkos::pair<IndexType, ValueType> goldPair,
                                   const ItType result,
                                   TestedViewType testedView) {
   // check that iterator is pointing to right element
+  // FIXME_OPENMPTARGET - With LLVM/13 we need PR 4456 to be merged. Else the
+  // following line fails duirng compile time.
```
Comment (dalg24) [inline algorithms/unit_tests/TestStdAlgorithmsMinMaxElementOps.cpp:206]:
```suggestion
  // following line fails during compile time.
```
or "at"

File: algorithms/unit_tests/TestStdAlgorithmsMinMaxElementOps.cpp (hunk 3 above)
Comment (rgayatri23) [reply algorithms/unit_tests/TestStdAlgorithmsMinMaxElementOps.cpp:209]: Yes. Apparently I cannot avoid this with an #ifdef. This needs #4456 . That will fix this compile time issue.

Comment (masterleinad) [reply algorithms/unit_tests/TestStdAlgorithmsMinMaxElementOps.cpp:209]: You can always do
```
#ifndef KOKKOS_ENABLE_OPENMPTARGET
  EXPECT_EQ(result - KE::begin(testedView), goldPair.first);
#else
  (void) testedView;
#endif
```

Comment (rgayatri23) [reply algorithms/unit_tests/TestStdAlgorithmsMinMaxElementOps.cpp:209]: I have tried that but looks like the CI still checks that line and fails at compile time. I was expecting the other PR to get merged before this and then we would not need the additional #ifdef.

**Comment by dalg24** [comment]
#4456 has been merged.  Would you please rebase.

File: core/unit_test/TestMathematicalSpecialFunctions.hpp [hunk 5]
```diff
@@ -1857,10 +1857,14 @@ TEST(TEST_CATEGORY, mathspecialfunc_expint1) {
   test.testit();
 }
 
+// FIXME_OPENMPTARGET: This unit test fails with a misaligned address error at
+// runtime with LLVM/13.
```
Comment (dalg24) [inline core/unit_test/TestMathematicalSpecialFunctions.hpp:1861]: Is it fixed in the development version?  If not did you report the issue?

**Comment by rgayatri23** [comment]
No. This is not yet fixed. The issue has been reported and is being looked into.

**Review by dalg24 — APPROVED** [review]

**Comment by rgayatri23** [comment]
> No. This is not yet fixed. The issue has been reported and is being looked into.

Just FYI, this issue is also being tracked here #4224

**Comment by dalg24** [comment]
Need to fix `error while loading shared libraries: libomp.so: cannot open shared object file: No such file or directory`

**Comment by crtrott** [comment]
Retest this please.

**Review by crtrott — APPROVED** [review]

**Review by dalg24 — CHANGES_REQUESTED** [review]
CI not passing

**Review by dalg24 — APPROVED** [review]

**Review by dalg24 — APPROVED** [review]
